feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -342,16 +341,11 @@ bool target_radio_config_set2(const struct schema_Wifi_Radio_Config *rconf,
 			 LOGE("%s: failed to set ht/hwmode", rconf->if_name);
 	}
 
-	list_channels_len = phy_get_list_channels_dfs(phy, list_channels);
-	if (list_channels_len) {
-		struct blob_attr *n;
-		n = blobmsg_open_array(&b, "channels");
-		for(int i = 0; i < list_channels_len; i++)
-		{
-			blobmsg_add_u32(&b, NULL, list_channels[i]);
-		}
-		blobmsg_close_array(&b, n);
-	}
+	struct blob_attr *n;
+	n = blobmsg_open_array(&b, "channels");
+	blobmsg_add_u32(&b, NULL, rconf->channel);
+	blobmsg_add_u32(&b, NULL, rrm_get_backup_channel(rconf->freq_band));
```

Comment:
Does it matter that this is now an option instead of a list with 1 entry?

The call to rrm_get_backup_channel() can fail returning 0.  What happens in hostapd when we set this channel to 0?  Previously, we had 'channels' set the same as 'channel' due to scripting.  Would that be a safer option in this case.

Reply:
@RickSommerville  I can add a check before writing the backup channel that its not 0, by providing the current channel alone in the list we would limit the AP from switching in case of radar event,